pkg/ui/spa_test.go

```diff
@@ -37,22 +39,40 @@ func (m *mockFileSystem) Stat(name string) (os.FileInfo, error) {
 	return nil, nil
 }
 
-type mockHttpFileSystem struct{}
+type mockHTTPFileSystem struct{}
 
-func (f mockHttpFileSystem) Open(name string) (http.File, error) {
+func (f mockHTTPFileSystem) Open(name string) (http.File, error) {
 	appFS := afero.NewMemMapFs()
-	appFS.MkdirAll("/ui/build", 0755)
-	appFS.MkdirAll("/ui/build/data/", 0755)
-	afero.WriteFile(appFS, "/ui/build/data/services.json", []byte("file a"), 0644)
-	afero.WriteFile(appFS, "/ui/build/index.html", []byte("file b"), 0644)
+	err := appFS.MkdirAll("/ui/build", 0755)
+	if err != nil {
```

Comment:
common/idiomatic go style for this would be:
```
if err := appF.MkdirAll("/ui/build", 0755); err != nil {
    return err
}
```

Do that throughout your changes.

Reply:
Beautiful, thanks very much! That just saved me 20 lines hahaha